allennlp/training/model.py

```diff
@@ -13,9 +13,9 @@ class Model(torch.nn.Module):
     data generation, because model inputs are named.
     """
 
-    def forward(self,
+    def forward(self,  # pylint: disable=arguments-differ
                 inputs: Dict[str, torch.Tensor],
-                compute_loss: bool = False):  # pylint: disable=arguments-differ
+                compute_loss: bool = False) -> (Union[torch.Tensor, List[torch.Tensor]], Optional[float]):
```

Comment:
This would be easy actually, just changing `Instance` and `Dataset`. We can leave the fields untouched.